### DISCUSSION TITLE
Escaping $ In Arrays?

### DISCUSSION THREAD

**Discussion by Lozoute** [vitaly-t/pg-promise#893] [Q&A] [answered]
Hi!

I'm using pg-promise to insert many values with only one query, with something like that:
 

```
const fields = ['externalId', 'skills']
const cs = new pgp.helpers.ColumnSet(fields, { table: 'my_table' })
const values = [{ externalId: 'Foo', skills: ['Toshop', 'InDesign'] }, { externalId: 'Bar', skills: ['unescaped$Dollars', 'Excel'] }]
const insertQuery = pgp.helpers.insert(values, cs)
```

(`externalId` is defined as `TEXT` and `skills` is defined as `TEXT[] DEFAULT NULL`)

But it generates an error as the $ is not cleaned by pg-promise.
So far I've been cleaning dollars myself with a regexp before passing the values to pg-promise.
Am I missing something? Is there a way to format a string or an array of string to automatically escape those wrong values?

Thanks!

**Comment by vitaly-t** [comment, marked as answer]
Why are you expecting for the dollar to be escaped? It doesn't need to be.

I ran that exact code, and there was no error in the code generated:

```sql
INSERT INTO "my_table"("externalId","skills") VALUES('Foo',ARRAY['Toshop','InDesign']),('Bar',ARRAY['unescaped$Dollars','Excel'])
```

So I don't know what error or issue you are talking about.

**Comment by Lozoute** [comment]
Hello, yes my bad, the problem was due to the way JavaScript handle dollar when present in the string replacing another with str.replace() function.

Thanks for the quick answer and your patience
